tasks/configure.yml

```diff
@@ -122,6 +122,42 @@
   register: postgresql_configuration_pt3
   changed_when: "'_OK_' not in postgresql_configuration_pt3.stdout"
 
+- name: PostgreSQL | Generate SSL server.key | RedHat
+  shell: "umask 0077; openssl genrsa 2048 > {{ postgresql_data_directory}}/server.key"
+  args:
+    creates: "{{ postgresql_data_directory }}/server.key"
+  become: true
+  become_user: "{{ postgresql_service_user }}"
+  when: ansible_os_family == "RedHat" and
+        postgresql_ssl == "on"
+
+- name: PostgreSQL | Secure SSL server.key | RedHat
+  file:
+    path: "{{ postgresql_data_directory }}/server.key"
+    owner: "{{ postgresql_service_user }}"
+    group: "{{ postgresql_service_group }}"
+    mode: 0600
+  when: ansible_os_family == "RedHat" and
+        postgresql_ssl == "on"
```

Comment:
postgresql_ssl is defined as a Boolean in https://github.com/ANXS/postgresql/blob/1d4686fdc2be0133e39601aef4fae4114c867f78/defaults/main.yml#L125
Here you use it as a string in the when: conditional.
Thus the conditional silently does not do what it is intended to do.
The conditional should be:
`when: ansible_os_family == "RedHat"  and  postgresql_ssl`

Same holds true for the similar conditionals below.

Reply:
Good point @edgecode , thanks.  I'll make sure it's testing a real boolean.